retrofit/src/main/java/retrofit/Utils.java

```diff
@@ -39,6 +39,12 @@
     return object;
   }
 
+  static boolean methodRequiresRequestBody(final String method) {
```

Comment:
The problem is that okhttp will not allow you to make a custom request with body...
You'll get
```
java.lang.IllegalArgumentException: method CUSTOM must not have a request body.
	at com.squareup.okhttp.Request$Builder.method(Request.java:241)
```
because its `permitsRequestBody` will return false.